examples/model_building/plot_bcic_iv_2a_moabb_trial_eegprep.py

```diff
@@ -0,0 +1,383 @@
+""".. _bcic-iv-2a-moabb-trial-eegprep:
+
+Basic Brain Decoding on EEG Data using EEGPrep preprocessor and EEGNet
+======================================================================
+
+This is a variant of the basic _moabb_trial decoding example that
+additionally inserts an EEGPrep stage into the preprocessing pipeline
+as a minimal demonstration of how to use EEGPrep with Braindecode. See
+the documentation of EEGPrep for more details on its functionality.
+
+.. contents:: This example covers:
+   :local:
+   :depth: 2
+
+"""
+
+######################################################################
+# Loading and preparing the data
+# -------------------------------------
+#
+
+
+######################################################################
+# Loading the dataset
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# First, we load the data. In this tutorial, we load the BCI Competition
+# IV 2a data [1]_ using braindecode's wrapper to load via
+# `MOABB library <moabb_>`_ [2]_.
+#
+# .. note::
+#    To load your own datasets either via mne or from
+#    preprocessed X/y numpy arrays, see :ref:`mne-dataset-example`
+#    and :ref:`custom-dataset-example`.
+#
+
+from braindecode.datasets import MOABBDataset
+
+subject_id = 3
+dataset = MOABBDataset(dataset_name="BNCI2014_001", subject_ids=[subject_id])
+
+
+######################################################################
+# Preprocessing
+# ~~~~~~~~~~~~~
+#
+
+
+######################################################################
+# Now we apply preprocessing like bandpass filtering to our dataset.
+# You can either apply functions provided by :class:`mne.io.Raw` or
+# :class:`mne.Epochs` or apply your own functions, either to the
+# MNE object or the underlying numpy array.
+#
+# .. note::
+#    Generally, braindecode prepocessing is directly applied to the loaded
+#    data, and not applied on-the-fly as transformations, such as in
+#    PyTorch-libraries like `<torchvision_>`_.
+#
```

Comment:
can you write a little more about the eeg prep itself, I think it is necessary here